redash/handlers/dashboards.py

```diff
@@ -146,7 +146,14 @@ def get(self, dashboard_slug=None):
         :>json string widget.created_at: ISO format timestamp for widget creation
         :>json string widget.updated_at: ISO format timestamp for last widget modification
         """
-        dashboard = get_object_or_404(models.Dashboard.get_by_slug_and_org, dashboard_slug, self.current_org)
+        if dashboard_slug.split('-')[0].isdigit():
+            identifier = int(dashboard_slug.split('-')[0])
+            fn = models.Dashboard.get_by_id_and_org
```

Comment:
We have 185 dashboards in our database that match this pattern already ("28 Days Growth Metrics"), so old URLs will stop working. We need a fallback here.

Some ideas:

* Try with `get_by_slug_and_org` if `get_by_id_and_org` fails.
* Test if the slug matches the dashboards slug.

Reply:
Preferred and implemented the first, but then realized that it breaks when you have a dashboard called "28 Days Growth Metrics" (available through `/dashboards/28-days-growth-metrics`) and then you create another dashboard called "I like trains" which gets assigned with id 28 (available through `/dashboards/28-i-like-trains`).

In that case, `get_by_id_and_org` wouldn't fail, and your previous links would break.

Reply:
Why?

When you request `28-days...` it will work with `get_by_slug_and_org`, and when you request `28-i-like...` it will fail `get_by_slug_and_org` and then try `get_by_id_and_org` which will work.

Reply:
Hehe, that's the opposite direction of what you suggested in the first option :)
Yeah, that might work, but it would add an extra query for _most_ dashboard loads. I went ahead and implemented the second option which seems to work fine.

Reply:
Rethinking it, there's no possible way I can think that we can remain agnostic to the slug if we don't do `get_by_slug_and_org` and fallback to `get_by_id_and_org` (i.e. if we don't go down this path, renaming dashboards would always break existing links). aba6e3b

Reply:
> Hehe, that's the opposite direction of what you suggested in the first option :)

:flushed: when I first read your reply I thought that's what happened, *read my suggestion again* but still parsed it in reverse order. :man_facepalming: 

> Yeah, that might work, but it would add an extra query for _most_ dashboard loads. I went ahead and implemented the second option which seems to work fine.

:thinking: how about the following:

Currently the dashboard URLs are `/dashboard/<slug>`. But they should actually be `/dashboards/<id>-<whatever>`. So what if as part of this change we will change the URLs and keep the old ones for backward compatibility. 

This way if we get a request for `/dashboard/...` we know it's an old URL and we should pass a slug but if it's `/dashboards/...` we use the new logic?

Reply:
👍 a851402 😈